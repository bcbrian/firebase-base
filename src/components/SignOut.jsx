import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';

export default class SignOut extends Component {
  handleClick = () => {
    this.props.onSignOut();
  };

  render() {
    return (
      <Button variant="contained" color="primary" onClick={this.handleClick}>
        Sign Out
      </Button>
    );
  }
}

SignOut.propTypes = {
  onSignOut: PropTypes.func.isRequired
};
