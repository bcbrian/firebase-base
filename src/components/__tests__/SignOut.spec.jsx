import React from 'react';
import renderer from 'react-test-renderer';
import SignOut from '../SignOut';

describe('SignOut Component', () => {
  it('should render sign up', () => {
    const onSignOut = jest.fn();
    const SignOutComp = renderer.create(<SignOut onSignOut={onSignOut} />);
    expect(SignOutComp.toJSON()).toMatchSnapshot();
  });

  it('should call the onSignOut prop function', () => {
    const onSignOut = jest.fn();
    const SignOutComp = renderer.create(<SignOut onSignOut={onSignOut} />);
    SignOutComp.getInstance().handleClick();
    expect(onSignOut).toHaveBeenCalledTimes(1);
    expect(SignOutComp.toJSON()).toMatchSnapshot();
  });
});
