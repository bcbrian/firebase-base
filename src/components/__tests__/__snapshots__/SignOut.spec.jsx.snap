// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut Component should call the onSignOut prop function 1`] = `
<div>
  <h3
    className="ui dividing header"
  >
    Sign Out
  </h3>
  <button
    className="ui fluid primary button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    Sign Out
  </button>
</div>
`;

exports[`SignOut Component should render sign up 1`] = `
<div>
  <h3
    className="ui dividing header"
  >
    Sign Out
  </h3>
  <button
    className="ui fluid primary button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    Sign Out
  </button>
</div>
`;
